hdc/algo/dekad.py

```diff
@@ -79,6 +79,14 @@ def __eq__(self, other: object) -> bool:
             return self._dkd == other._dkd
         return False
 
+    def __lt__(self, other: object) -> bool:
+        """Check for less than inequality with other Dekad or a string."""
+        if isinstance(other, str):
+            return str(self) < other
+        if isinstance(other, Dekad):
+            return self._dkd < other._dkd
+        return False
```

Comment:
Should we add a `datetime` comparison as well? Returning `True` when `datetime` object is to the left of `.start_date`?